extensions/eda/plugins/event_source/kafka.py

```diff
@@ -51,19 +58,26 @@ async def main(  # pylint: disable=R0914
     group_id = args.get("group_id", None)
     offset = args.get("offset", "latest")
     encoding = args.get("encoding", "utf-8")
+    security_protocol = args.get("security_protocol", "PLAINTEXT")
+    sasl_mechanism = args.get("sasl_mechanism", "PLAIN")
+    sasl_plain_username = args.get("sasl_plain_username")
+    sasl_plain_password = args.get("sasl_plain_password")
 
     if offset not in ("latest", "earliest"):
         msg = f"Invalid offset option: {offset}"
         raise ValueError(msg)
 
-    if cafile:
-        context = create_ssl_context(
+    ssl_context = None
+    if cafile or security_protocol.endswith("SSL"):
+        if security_protocol == "PLAINTEXT":
```

Comment:
`security_protocol` should be `SASL_SSL` when `cafile` is provided and `security_protocol` is `SASL_PLAINTEXT`.